app/helpers/application_helper.rb

```diff
@@ -36,4 +36,8 @@ def service_link
   def current_namespace
     params[:controller].split('/').first
   end
+
+  def intended_namespace
```

Comment:
Looking at this, shall we just change the implementation of `current_namespace`? Bit confusing to have two methods that return the same thing in all circs except 404 pages

Reply:
I've started using `intended_namespace` for the footer, too, which will display the relevant footers for candidate and provider error pages e.g.

![image](https://user-images.githubusercontent.com/107591/89556400-3ee93e80-d809-11ea-8749-ce13b9946c8e.png)

The only thing stopping us from turning `intended_namespace` into the new `current_namespace` is, ironically, the lack of custom provider error pages. This is because the header component will then start displaying the sub-navigation items, but the error page lacks a user session context. For example (for a signed-in provider user):

![image](https://user-images.githubusercontent.com/107591/89556700-b0c18800-d809-11ea-9cb7-23df39b210c8.png)

How about we merge this as is (with both `current_namespace` and `intended_namespace`) and merge the two methods in another PR providing interface-specific error controllers and pages (so that they get access to `current_candidate`, `current_provider_user` and `current_support_user`)?